.evergreen/config.yml

```diff
@@ -212,6 +212,55 @@ functions:
     - command: expansions.update
       params:
         file: src/serverless-expansion.yml
+    - command: shell.exec
+      params:
+        shell: "bash"
+        script: |
+          ${PREPARE_SHELL}
+          
+          if [ -z "${SERVERLESS_MONGODB_VERSION}" ]; then
+            echo "expected SERVERLESS_MONGODB_VERSION to be set"
+            exit 1
+          fi
+          
+          # Download the enterprise server download for current platform to $MONGODB_BINARIES.
+          # This is required for tests that need mongocryptd.
+          # $MONGODB_BINARIES is added to the $PATH in fetch-source.
+          ${PYTHON3_BINARY} $DRIVERS_TOOLS/.evergreen/mongodl.py \
+              --component archive \
+              --version ${SERVERLESS_MONGODB_VERSION} \
+              --edition enterprise \
+              --out $MONGODB_BINARIES \
+              --strip-path-components 2
+          
+          # Download the crypt_shared dynamic library for the current platform.
```

Comment:
IIRC, this is handled by [run-orchestration.sh](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/2443440ca69bcdba825eb0b43f874006ae3b5bca/.evergreen/run-orchestration.sh#L105) in drivers-evergreen-tools for non-serverless tests. That appears to be exactly what [DRIVERS-2459](https://jira.mongodb.org/browse/DRIVERS-2459) is meant to address.

IMO, this should have been addressed as a prerequisite for [DRIVERS-2400](https://jira.mongodb.org/browse/DRIVERS-2400) before spawning language tickets and inviting drivers to sort this out on their own.